frontend/public/components/nav/nav-header.tsx

```diff
@@ -16,20 +19,50 @@ export type NavHeaderProps = {
   onPerspectiveSelected: () => void;
 };
 
+export const ACM_LINK_ID = 'acm-console-link';
+const PerspectiveURLParam = 'perspective';
+const PerspectiveParamValues = ['dev', 'admin'];
+
+const getACMConsoleLink = (links: K8sResourceKind[]): K8sResourceKind =>
+  links.find(
+    (link) => link.spec.location === 'ApplicationMenu' && link.metadata.name === ACM_LINK_ID,
+  );
+
 const NavHeader_: React.FC<NavHeaderProps & StateProps> = ({ onPerspectiveSelected, flags }) => {
   const [activePerspective, setActivePerspective] = useActivePerspective();
   const [isPerspectiveDropdownOpen, setPerspectiveDropdownOpen] = React.useState(false);
   const perspectiveExtensions = useExtensions<Perspective>(isPerspective);
+  const [consoleLinks] = useK8sWatchResource<K8sResourceKind[]>({
+    isList: true,
+    kind: referenceForModel(ConsoleLinkModel),
+    optional: true,
+  });
+  const acmLink: K8sResourceKind = getACMConsoleLink(consoleLinks);
   const togglePerspectiveOpen = React.useCallback(() => {
     setPerspectiveDropdownOpen(!isPerspectiveDropdownOpen);
   }, [isPerspectiveDropdownOpen]);
 
+  const urlParams = new URLSearchParams(window.location.search);
+  const perspectiveParam = urlParams.get(PerspectiveURLParam);
+  if (
+    perspectiveParam &&
+    PerspectiveParamValues.includes(perspectiveParam) &&
+    perspectiveParam !== activePerspective
+  ) {
+    setActivePerspective(perspectiveParam);
+  }
```

Comment:
I think this is the wrong place for this logic. I'm not too familiar with the code, but probably you want it in PerspectiveDetector.tsx.

Reply:
Ok, moved to `frontend/packages/console-app/src/components/detect-perspective/DetectPerspective.tsx`